goal_src/jak2/engine/common_objs/projectile.gc

```diff
@@ -7,3 +7,720 @@
+
+       If we've met or exceeded the projectiles maximum allowed hits, switch to the [[projectile::impact]] state"
+  (when (-> obj attack-mode)
+    (let ((a2-1 (-> arg1 param 0)))
+      (when (deal-damage! obj arg0 (the-as event-message-block a2-1))
+        (let ((v1-3 (-> obj notify-handle)))
+          (send-event (handle->process v1-3) 'notify 'attack arg0)
+          )
+        (+! (-> obj hits) 1)
+        (if (and (>= (-> obj hits) (-> obj max-hits))
+                 (not (and (-> obj next-state) (= (-> obj next-state name) 'impact)))
+                 )
+            (go (method-of-object obj impact))
+            )
+        )
+      )
+    )
+  )
+
+(defmethod kill-projectile! projectile ((obj projectile))
+  "Transition to the [[projectile::impact]] state, always return [[#t]]"
+  (if (not (and (-> obj next-state) (= (-> obj next-state name) 'impact)))
+      (go (method-of-object obj impact))
+      )
+  #t
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod draw-laser-sight projectile ((obj projectile))
+  "TODO - confirm If applicable, draw the laser sight particles
+       :virtual"
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod spawn-impact-particles projectile ((obj projectile))
+  "Spawns associated particles with the projectile if applicable"
+  (if (nonzero? (-> obj part))
+      (sparticle-launch-control-method-11 (-> obj part) (-> obj root-override trans))
+      )
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod spawn-shell-particles projectile ((obj projectile))
+  "TODO - confirm"
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod unknown-particles projectile ((obj projectile))
+  "TODO - confirm"
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs sound-id.
+(defmethod play-impact-sound projectile ((obj projectile) (arg0 projectile-options))
+  (the-as sound-id 0)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod stop-sound! projectile ((obj projectile))
+  "Stops the current `sound-id` if set, re-init the `sound-id` after being stopped"
+  (when (nonzero? (-> obj sound-id))
+    (sound-stop (-> obj sound-id))
+    (set! (-> obj sound-id) (new 'static 'sound-id))
+    0
+    )
+  0
+  (none)
+  )
+
+(defmethod made-impact? projectile ((obj projectile))
+  "TODO - queries the collision cache, return true/false"
+  (let ((v1-0 (-> obj root-override)))
+    (let ((a1-0 (new 'stack-no-clear 'collide-query)))
+      (set! (-> a1-0 radius) (-> v1-0 root-prim prim-core world-sphere w))
+      (set! (-> a1-0 collide-with) (-> v1-0 root-prim prim-core collide-with))
+      (set! (-> a1-0 ignore-process0) obj)
+      (set! (-> a1-0 ignore-process1) (handle->process (-> obj ignore-handle)))
+      (set! (-> a1-0 ignore-pat) (-> v1-0 pat-ignore-mask))
+      (set! (-> a1-0 action-mask) (the-as uint 1))
+      )
+    (collide-shape-moving-method-65 v1-0 (-> v1-0 transv) -977272832 0 -989855744)
```

Comment:
Maybe the weird `-977272832` is actually a `float`? In jak 2 we can't tell, can you try changing `collide-shape-moving-method-65` to take `float`s and see if it looks better?